batch/batch/front_end/front_end.py

```diff
@@ -540,12 +540,9 @@ def check_service_account_permissions(user, sa):
         return
     if user == 'ci':
         if sa['name'] in ('ci-agent', 'admin'):
-            if DEFAULT_NAMESPACE == 'default':  # real-ci needs access to all namespaces
-                return
-            if sa['namespace'] == BATCH_PODS_NAMESPACE:
-                return
-    if user == 'test':
-        if sa['name'] == 'test-batch-sa' and sa['namespace'] == BATCH_PODS_NAMESPACE:
+            return
```

Comment:
I think the new rule should be this:

```
if user == 'ci':
    if sa['name'] in ('ci-agent', 'admin'):
        if DEFAULT_NAMESPACE == 'default' or sa['namespace'] == DEFAULT_NAMESPACE:
              return
```

The rationale was that ci needs to be able to access all namespaces, but if it's a test version of ci, it should only have access to its own namespace.

i.e. ci in the default namespace can use a service account in `test-pr-XXX-default` or in `default`, but the ci in the namespace `test-pr-XXX-default` should only be able to access `test-pr-XXX-default` and not `default`.